src/runtime/set.cpp

```diff
@@ -125,26 +148,61 @@ Box* frozensetNew(Box* _cls, Box* container) {
     return makeNewSet(cls, container);
 }
 
-Box* setNew(Box* _cls, Box* container) {
+Box* setNew(Box* _cls, BoxedTuple* args, BoxedDict* kwargs) {
     RELEASE_ASSERT(_cls->cls == type_cls, "");
     BoxedClass* cls = static_cast<BoxedClass*>(_cls);
     RELEASE_ASSERT(isSubclass(cls, set_cls), "");
+    int args_sz = args->size();
+
+    // only check set_cls itself, the subclass of set could take keyword argument
+    // same as __init__, please see the disscusstion of CPython bug #1486663
+    if (_cls == set_cls && kwargs) {
+        raiseExcHelper(TypeError, "set() does not take keyword arguments");
+    }
+
+    if (args_sz > 1)
+        raiseExcHelper(TypeError, "set() takes at most 1 argument (%d given)", args_sz);
 
     // Note: set.__new__ explicitly ignores the container argument.
     return makeNewSet(cls, NULL);
 }
 
-Box* setInit(Box* _self, Box* container) {
+Box* setInit(Box* _self, BoxedTuple* args, BoxedDict* kwargs) {
     RELEASE_ASSERT(PySet_Check(_self), "");
 
+    if (_self->cls == set_cls && kwargs) {
+        raiseExcHelper(TypeError, "set() does not take keyword arguments");
+    }
+
+    int args_sz = args->size();
```

Comment:
Could we keep this as a single optional positional argument instead of taking varargs?